### DISCUSSION TITLE
Making oil.nvim work with autoload

### DISCUSSION THREAD

**Discussion by alankritjoshi** [olimorris/persisted.nvim#115] [Q&A] [answered]
I am using lazy vim and oil.nvim.

With oil enabled, autoload doesn't work, it just opens oil buffer. I have to manually load session then. 

If I don't use oil, netrw opens and autoload works fine. 

Any idea what I am doing wrong?

Ref: 

- [persisted config](https://github.com/alankritjoshi/dotfiles/blob/f9abb14ffb8291eddd27bc1c98b940f49995a61c/private_dot_config/nvim/lua/plugins/persisted.lua)
- [oil config](https://github.com/alankritjoshi/dotfiles/blob/f9abb14ffb8291eddd27bc1c98b940f49995a61c/private_dot_config/nvim/lua/plugins/oil.lua)
- [autocmds](https://github.com/alankritjoshi/dotfiles/blob/f9abb14ffb8291eddd27bc1c98b940f49995a61c/private_dot_config/nvim/lua/config/autocmds.lua)

**Comment by alankritjoshi** [comment]
> Autoloading will not occur if the plugin is lazy loaded or a user opens Neovim with arguments other than a single directory argument. For example: nvim some_file.rb will not result in autoloading but nvim some/existing/path or nvim . will.

I noticed that autoloading works if I just do `nvim` but not if I do `nvim .`

**Comment by olimorris** [comment, marked as answer]
The README contains an example of how you can customise autoloading to suit your own needs.
